code-studio/src/js/callouts.js

```diff
@@ -112,16 +124,49 @@ module.exports = function createCallouts(callouts) {
     }
 
     if (callout.on) {
-      $(window).on(callout.on, function () {
-        if (!callout.seen && $(selector).length > 0) {
+      $(window).on(callout.on, function (e, action) {
+        if (!config.codeStudio.selector) {
+          config.codeStudio.selector = selector;
+        }
+        var lastSelector = config.codeStudio.selector;
+        $(window).trigger('prepareforcallout', [config.codeStudio]);
+        if (lastSelector !== config.codeStudio.selector && $(lastSelector).length > 0) {
+          $(lastSelector).qtip(config).qtip('destroy');
+        }
+        if ($(config.codeStudio.selector).length > 0) {
+          if (action === 'hashchange' || action === 'hashinit' || !callout.seen) {
+            $(config.codeStudio.selector).qtip(config).qtip('show');
+          }
           callout.seen = true;
-          $(selector).qtip(config).qtip('show');
         }
       });
-    } else {
+    } else if (!callout.seen) {
       $(selector).qtip(config).qtip('show');
     }
   });
+
+  // Insert a hashchange handler to detect triggercallout= hashes and fire
+  // appropriate events to open the callout
+  function detectTriggerCalloutOnHash(event) {
```

Comment:
So authoring in my current proposal looks like this:
```
You need to [draw sprites](#triggercallout=drawSpritesCallout) and
```
How does one author that URL above in markdown? I assume you'd have to use the more traditional HTML syntax and not markdown, correct? I'm not opposed to your change, but it may make the authoring harder to clean up our side. But since we're not using hash tags for anything yet (many "apps" on the web use it extensively with full paths after the hash).
Also, how does one attach to the link event in this case?